packages/electron-updater/src/MacUpdater.ts

```diff
@@ -79,12 +80,26 @@ export class MacUpdater extends AppUpdater {
       throw newError(`ZIP file not provided: ${safeStringifyJson(files)}`, "ERR_UPDATER_ZIP_FILE_NOT_FOUND")
     }
 
+    const provider = downloadUpdateOptions.updateInfoAndProvider.provider
+
     return this.executeDownload({
       fileExtension: "zip",
       fileInfo: zipFileInfo,
       downloadUpdateOptions,
-      task: (destinationFile, downloadOptions) => {
-        return this.httpExecutor.download(zipFileInfo.url, destinationFile, downloadOptions)
+      task: async (destinationFile, downloadOptions) => {
+        const currentVersionFileName = "currentVersion"
+        const currentVersion = readFileSync(currentVersionFileName, "utf8")
+        let differentialDownloadedFailed = false
+        if (currentVersion === this.app.version) {
```

Comment:
Can you describe why we're checking if the current version is the same as the currently installed version? I noticed in `NsisUpdater.ts` that it's simply running the following without that `currentVersion` string check
```
        if (isWebInstaller || (await this.differentialDownloadInstaller(fileInfo, downloadUpdateOptions, destinationFile, provider))) {
          await this.httpExecutor.download(fileInfo.url, destinationFile, downloadOptions)
        }
```

Reply:
During nsis installation, the executable file is copied to the updaterCacheDirName directory. During incremental updates, the application can find itself in this directory and proceed with the update.

<img width="1678" alt="image" src="https://github.com/electron-userland/electron-builder/assets/5462444/56b7c20c-ab17-46f4-a962-8575c3c95dab">


However, the same functionality has not been implemented for Mac's pkg package installation. I'm currently unsure if this functionality can be achieved for Mac. One approach could be to create a zip archive of the entire app directory in the postinstall script and move it to the updaterCacheDirName directory.

Currently, my approach is as follows: 
After fresh installation, then the first update, as there is no zip package for the current version, a full update is performed. Once the zip package is downloaded, it's copied to the updaterCacheDirName directory. From the second update onwards, incremental updates can be performed. 
However, if a user manually performs an overwrite installation in future updates, a mismatch might occur between the updater's zip package and the current version's package. To avoid this, I'm maintaining a record of the current version in a file. I acknowledge that this approach isn't as elegant as NSIS, and I'm working on a solution similar to NSIS.